sql-native/src/main/java/org/ballerinalang/sql/parameterprocessor/AbstractResultParameterProcessor.java

```diff
@@ -0,0 +1,189 @@
+/*
+ *  Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+package org.ballerinalang.sql.parameterprocessor;
+
+import io.ballerina.runtime.api.types.Field;
+import io.ballerina.runtime.api.types.StructureType;
+import io.ballerina.runtime.api.types.Type;
+import io.ballerina.runtime.api.values.BArray;
+import io.ballerina.runtime.api.values.BObject;
+import io.ballerina.runtime.api.values.BString;
+import org.ballerinalang.sql.exception.ApplicationError;
+import org.ballerinalang.sql.utils.ColumnDefinition;
+
+import java.math.BigDecimal;
+import java.sql.Array;
+import java.sql.Blob;
+import java.sql.CallableStatement;
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.SQLXML;
+import java.sql.Statement;
+import java.sql.Struct;
+import java.util.List;
+
+/**
+ * This class has abstract implementation of methods required convert SQL types into ballerina types and
+ * other methods that process the parameters of the result.
+ *
+ * @since 0.5.6
+ */
+public abstract class AbstractResultParameterProcessor {
+
+    protected abstract BArray createAndPopulateCustomValueArray(Object firstNonNullElement, Object[] dataArray)
+            throws ApplicationError;
+
+    protected abstract BArray createAndPopulateCustomBBRefValueArray(Object firstNonNullElement, Object[] dataArray,
+                                                                     Type type) throws ApplicationError;
+
+    protected abstract void createUserDefinedTypeSubtype(Field internalField, StructureType structType)
+            throws ApplicationError;
+
+    protected abstract BArray convert(Array array, int sqlType, Type type) throws SQLException, ApplicationError;
```

Comment:
Shall we rename these? Instead of just convert maybe `getBArray`  something else like that? Also make sure all the functions follow the same naming standards

Reply:
Resolved with a5e57ba